lib/topics_filter.rb

```diff
@@ -1,67 +1,52 @@
 # frozen_string_literal: true
 
 class TopicsFilter
-  def initialize(guardian:, scope: Topic)
+  attr_reader :topic_notification_levels
+
+  def initialize(guardian:, scope: Topic.all)
     @guardian = guardian
     @scope = scope
-    @notification_levels = Set.new
-    @registered_scopes = {}
+    @topic_notification_levels = Set.new
   end
 
-  def topic_notification_levels
-    @registered_scopes.dig(:topic_notification_levels, :params, :notification_levels) || []
-  end
+  FILTER_ALIASES = { "categories" => "category" }
+  private_constant :FILTER_ALIASES
 
   def filter_from_query_string(query_string)
     return @scope if query_string.blank?
 
+    filters = {}
+
     query_string.scan(
       /(?<key_prefix>[-=])?(?<key>[\w-]+):(?<value>[^\s]+)/,
     ) do |key_prefix, key, value|
-      case key
+      key = FILTER_ALIASES[key] || key
+
+      filters[key] ||= {}
```

Comment:
Minor but you can use `filters[key] ||= Hash.new([])`

In that case, init empty array will be not necessary  `filters[key]["key_prefixes"] ||= []` and `filters[key]["values"] ||= []`

<img width="317" alt="Screenshot 2023-04-13 at 2 10 47 pm" src="https://user-images.githubusercontent.com/72780/231650852-ff929725-e237-4a9d-bd90-43882fe64920.png">

Reply:
O nice suggestion :+1:

Reply:
Hmm I tried this but it didn't work for some reason.